components/usage/pkg/stripe/stripe.go

```diff
@@ -320,6 +320,62 @@ func (c *Client) GetSubscriptionWithCustomer(ctx context.Context, subscriptionID
 	return subscription, nil
 }
 
+func (c *Client) CreateSubscription(ctx context.Context, customerID string, priceID string) (*stripe.Subscription, error) {
+	if customerID == "" {
+		return nil, fmt.Errorf("no customerID specified")
+	}
+	if priceID == "" {
+		return nil, fmt.Errorf("no priceID specified")
+	}
+
+	params := &stripe.SubscriptionParams{
+		Customer: &customerID,
+		Items: []*stripe.SubscriptionItemsParams{
+			{
+				Price: &priceID,
+			},
+		},
+	}
+
+	subscription, error := c.sc.Subscriptions.New(params)
+	if error != nil {
+		return nil, fmt.Errorf("failed to get subscription with customer ID %s", customerID)
+	}
+
+	return subscription, error
+}
+
+func (c *Client) SetDefaultPaymentForCustomer(ctx context.Context, customerID string, setupIntentId string) (*stripe.Customer, error) {
+	if customerID == "" {
+		return nil, fmt.Errorf("no customerID specified")
+	}
+
+	if setupIntentId == "" {
+		return nil, fmt.Errorf("no setupIntentID specified")
+	}
+
+	setupIntent, err := c.sc.SetupIntents.Get(setupIntentId, &stripe.SetupIntentParams{
+		Params: stripe.Params{
+			Context: ctx,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to retrieve setup intent with id %s", setupIntentId)
+	}
+
+	paymentMethod, err := c.sc.PaymentMethods.Attach(setupIntent.PaymentMethod.ID, &stripe.PaymentMethodAttachParams{Customer: &customerID})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to attach payment method to setup intent ID %s", setupIntentId)
+	}
+
+	customer, _ := c.sc.Customers.Update(customerID, &stripe.CustomerParams{InvoiceSettings: &stripe.CustomerInvoiceSettingsParams{DefaultPaymentMethod: &paymentMethod.ID}, Address: &stripe.AddressParams{Line1: &paymentMethod.BillingDetails.Address.Line1, Country: &paymentMethod.BillingDetails.Address.Country}})
```

Comment:
[nit] This is fine, and go linter will accept it but I'd recommend spreading this over a few lines just to aid readability, especially in Github source browsing.